panels/site_sections/attractions_admin.py

```diff
@@ -324,6 +314,42 @@ def delete_event(self, session, id):
         if message:
             return {'error': message}
 
+    @ajax
+    def cancel_signup(self, session, id):
+        message = ''
+        if cherrypy.request.method == 'POST':
+            signup = session.query(AttractionSignup).get(id)
+            attraction_id = signup.event.feature.attraction_id
+            attraction = session.query(Attraction).get(attraction_id)
+            if not session.admin_attendee().can_admin_attraction(attraction):
+                message = "You cannot cancel a signup for an attraction you don't own"
+            elif signup.checkin_time:
+                message = "You cannot cancel a signup that has already checked in"
+            else:
+                session.delete(signup)
+                session.commit()
+        if message:
+            return {'error': message}
+
+    @only_renderable(c.STUFF, c.PEOPLE, c.REG_AT_CON)
+    def checkin(self, session, message='', **params):
+        id = params.get('id')
+        if not id:
+            raise HTTPRedirect('index')
+
+        try:
+            uuid.UUID(id)
+            filters = [Attraction.id == id]
+        except Exception:
+            filters = [Attraction.slug.startswith(sluggify(id))]
```

Comment:
Oh man, this is a SUPER useful feature. We don't have to do it for this PR, but we should totally make this into a decorator. Imagine if any model with a `slug` column could have pretty URLs instead of ID-based URLs!

Reply:
Yes, yes, yes! Nice URLs everywhere!